File: competitions/Week_of_Code_35/lucky_purchase.py
#https://www.hackerrank.com/contests/w35/challenges/lucky-purchase

def validPrice(price):
    sevens = 0
    fours = 0
    for c in price:
        if c == "7":
            sevens += 1
        elif c == "4":
            fours += 1
        else:
            return False
    return sevens == fours
            

if __name__ == "__main__":
    min_price = -1
    best_name = -1
    n = int(input())
    for i in range(n):
        line = input().strip().split()
        name = line[0]
        price = int(line[1])
        valid = validPrice(str(price))
        if valid:
            if min_price == -1:
                min_price = price
                best_name = name
            else:
                if price < min_price:
                    min_price = price
                    best_name = name
    print (best_name)
        
            

File: competitions/Week_of_Code_35/lucky_purchase.cpp
#include <bits/stdc++.h>
using namespace std;

bool validPrice(string price) {
	int sevens = 0;
	int fours = 0;

	for (int i = 0; i < price.length(); i++) {
		if (price[i] == '4') {
			fours += 1;
		}

		else if (price[i] == '7') {
			sevens += 1;
		}

		else {
			return false;
		}
	}
		return sevens == fours;
}

int main() {
	int n;
	int min_price = -1;
	string best_name = "-1";
	cin >> n;
	
	string name;
	string price;
	for(int i = 0; i < n; i++) {
		cin >> name;
		cin >> price;
		bool valid = validPrice(price);
		if (valid) {
			int p = atoi(price.c_str());
			if (min_price == -1) {
				min_price = p;
				best_name = name;
			}
			else {
				if (p < min_price) {
					min_price = p;
					best_name = name;
				}
			}
		}
	}
	cout << best_name;
	
	return 0;
}

